_posts/2023-04-13-denoising-diffusion/denoising-diffusion.Rmd

```diff
@@ -0,0 +1,199 @@
+likelihood of the data under the model. From the above-listed
+architectures, this is the case with VAEs and flows; it is not with
+GANs.
+
+But we can also take a different perspective -- that of purpose.
+Firstly, are we interested in representation learning? That is, would we
+like to condense the space of samples into a sparser one, one that
+exposes underlying features and gives hints at useful categorization? If
+so, VAEs are the classical candidates to look at.
+
+Alternatively, are we mainly interested in generation, and would like to
+synthesize samples corresponding to different levels of coarse-graining?
+Then diffusion algorithms are a good choice. It has been shown that
+
+> [...] representations learnt using different noise levels tend to
+> correspond to different scales of features: the higher the noise
+> level, the larger-scale the features that are captured.[^1]
+
+[^1]: From @dieleman2022diffusion, an excellent high-level introduction
+    to diffusion models.
+
+As a final example, what if we aren't interested in synthesis, but would
+like to assess if a given piece of data could likely be part of some
+distribution? If so, flows might be an option.[^2]
+
+[^2]: For more information on this lesser-used family of algorithms, see
+    <https://blog.evjang.com/2018/01/nf1.html>.
+
+## Zooming in: Diffusion models
+
+Just like about every deep-learning architecture, diffusion models
+constitute a heterogeneous family. Here, let us just name a few of the
+most en-vogue members. [^3]
+
+[^3]: En vogue as of this writing (the usual caveat that comes with
+    using epitheta such as this).
+
+When, above, we said that the idea of diffusion models was to
+sequentially transform an input into noise, then sequentially de-noise
+it again, we left open how that transformation is operationalized. This,
+in fact, is one area where rivaling approaches tend to differ.
+@abs-2011-13456, for example, make use of a a stochastic differential
+equation (SDE) that maintains the desired distribution during the
+information-destroying forward phase. In stark contrast, other
+approaches, inspired by @ho2020, rely on Markov chains to realize state
+transitions. The variant introduced here -- @song2020 -- keeps the same
+spirit, but improves on efficiency.
+
+## Our implementation -- overview
+
+The [README](https://github.com/dfalbel/denoising-diffusion) provides a
+very thorough introduction, covering (almost) everything from
+theoretical background via implementation details to training procedure
+and tuning. Here, we just outline a few basic facts.
+
+As already hinted at above, all the work happens during the forward
+stage. The network takes two inputs, the images as well as information
+about the signal-to-noise ratio to be applied at every step in the
+corruption process. (That information may be encoded in various ways,
+and is then embedded, in some form, into a higher-dimensional space more
+conducive to learning.)
```

Comment:
I think it would be nice to find a way to illustrate the forward diffusion. Maybe use this picture in between these paragraphs? 
![](https://github.com/dfalbel/denoising-diffusion/raw/main/README_files/figure-commonmark/reverse-1.png)

Reply:
will do!